Add resources

brramos opened the pull request:
- update acl security rule resource
- add data center resource
- add data center servers resource
- add incap rule resource
- update site resource

brramos commented on incapsula/resource_incap_rule.go:
```diff
@@ -0,0 +1,234 @@
+			"enabled": {
+				Description: "Is enabled",
+				Type:        schema.TypeString,
+				Required:    true,
```
will do

brramos commented on incapsula/resource_incap_rule.go:
```diff
@@ -0,0 +1,234 @@
+			},
+
+			// Optional Arguments
+			"site_id": {
```
updating to required

brramos commented on incapsula/resource_data_center_servers.go:
```diff
@@ -0,0 +1,139 @@
+	return nil
+}
+
+func resourceDataCenterServersDelete(d *schema.ResourceData, m interface{}) error {
```
Delete through api is not working as expected. Following the pattern of setting id to empty string to clear resource when site is deleted. Confirmed data center severs are released when site is deleted.

brramos replied:
let me look into this more. My example maybe wrong.

brramos replied:
updated example

brramos commented on incapsula/resource_data_center.go:
```diff
@@ -0,0 +1,142 @@
+}
+
+func resourceDataCenterDelete(d *schema.ResourceData, m interface{}) error {
+	// todo: review delete data center
```
Delete through api is not working as expected. Following the pattern of setting id to empty string to clear resource when site is deleted. Confirmed data center are released when site is deleted.

brramos replied:
let me look into this more. My example maybe wrong.

brramos replied:
updated example

brramos commented on incapsula/resource_incap_rule.go:
```diff
@@ -0,0 +1,234 @@
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+			"priority": {
```
🤔 required? I see in specs that majority have an asterisk for being required but I see a few where it is not required. Not sure to make this optional and with default value of "1"?

brramos commented on incapsula/resource_incapsula_site.go:
```diff
@@ -62,6 +62,51 @@ func resourceSite() *schema.Resource {
 				Type:        schema.TypeString,
 				Optional:    true,
 			},
+			"site_id": &schema.Schema{
```
removing

brramos commented on incapsula/resource_incapsula_site.go:
```diff
@@ -166,8 +212,112 @@ func resourceSiteRead(d *schema.ResourceData, m interface{}) error {
-	return nil
+	client := m.(*Client)
+
+	siteID := d.Get("site_id").(int)
```
updated

Techcadia commented:
@brramos Checking in to see if this might be ready for a merge?

trx35479 commented:
We are looking to use this provider, we need the features from this branch. Just wondering if it is good to merge or will be planning to merge in couple of weeks or so. I will be pulling this branch for now to test our use cases but wondering if there would be a proper release in coming weeks or you will be adding more features?

brramos commented:
I have addressed all pr comments @Techcadia on April 22. I have no access to test this code anymore.
The last change was making `enabled` an optional value with a default value of true in `resource_incap_rule.go`. My example terraform file was also updated to allow deletion of data center and data center servers.
